skopt/gp_opt.py

```diff
@@ -174,6 +187,14 @@ def gp_minimize(func, dimensions, base_estimator=None, alpha=10e-10,
     specs = {"args": copy.copy(inspect.currentframe().f_locals),
              "function": inspect.currentframe().f_code.co_name}
 
+    if callback is not None:
```

Comment:
Should we start the refactoring campaign for repeated stuff across the `*_minimize` functions in this PR?

This bit looks like an ideal candidate for: `callback = check_callback(callback)` where `check_callback` lives in `skopt.utils` or some such (inspired by `check_array` and friends in `sklearn`).